common/src/main/java/tachyon/Version.java

```diff
@@ -23,15 +28,43 @@
 public final class Version {
   public static final String VERSION;
 
+  private static Map<String, Set<Long>> sCompatibleVersions;
+
   static {
     TachyonConf tachyonConf = new TachyonConf();
     VERSION = tachyonConf.get(Constants.TACHYON_VERSION);
+
+    sCompatibleVersions = new HashMap<String, Set<Long>>();
```

Comment:
After our offline discussion, I got rid of this. Clients will be only compatible with a service that has a matching version.